### ISSUE TITLE
0.8.2. fails to build for Alpine

### ISSUE THREAD

**Issue by danilobuerger** [krallin/tini#22]
Build command is:

```
build() {
    cd "$_builddir"
    export CFLAGS="-DPR_SET_CHILD_SUBREAPER=36 -DPR_GET_CHILD_SUBREAPER=37"
    cmake . || return 1
    make || return 1
}
```

Error log is:

```
-- The C compiler identification is GNU 4.9.2
-- Check for working C compiler: /usr/bin/gcc
-- Check for working C compiler: /usr/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
fatal: Not a git repository (or any of the parent directories): .git
fatal: Not a git repository (or any of the parent directories): .git
-- Configuring done
-- Generating done
-- Build files have been written to: /build/src/tini-0.8.2
Scanning dependencies of target tiniScanning dependencies of target tini-static

[ 50%] [100%] Building C object CMakeFiles/tini.dir/src/tini.c.oBuilding C object CMakeFiles/tini-static.dir/src/tini.c.o

<command-line>:0:0: error: "_FORTIFY_SOURCE" redefined
<built-in>: note: this is the location of the previous definition
<command-line>:0:0: error: "_FORTIFY_SOURCE" redefined
<built-in>: note: this is the location of the previous definition
CMakeFiles/tini-static.dir/build.make:54: recipe for target 'CMakeFiles/tini-static.dir/src/tini.c.o' failed
make[2]: *** [CMakeFiles/tini-static.dir/src/tini.c.o] Error 1
CMakeFiles/Makefile2:95: recipe for target 'CMakeFiles/tini-static.dir/all' failed
make[1]: *** [CMakeFiles/tini-static.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
CMakeFiles/tini.dir/build.make:54: recipe for target 'CMakeFiles/tini.dir/src/tini.c.o' failed
make[2]: *** [CMakeFiles/tini.dir/src/tini.c.o] Error 1
CMakeFiles/Makefile2:60: recipe for target 'CMakeFiles/tini.dir/all' failed
make[1]: *** [CMakeFiles/tini.dir/all] Error 2
Makefile:136: recipe for target 'all' failed
make: *** [all] Error 2
>>> ERROR: tini: all failed
>>> tini: Uninstalling dependencies...
```

**Comment by danilobuerger** [comment]
Building 0.8.0 just gives a warning regarding `_FORTIFY_SOURCE`:

```
-- The C compiler identification is GNU 4.9.2
-- Check for working C compiler: /usr/bin/gcc
-- Check for working C compiler: /usr/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
fatal: Not a git repository (or any of the parent directories): .git
fatal: Not a git repository (or any of the parent directories): .git
-- Configuring done
-- Generating done
-- Build files have been written to: /build/src/tini-0.8.0
Scanning dependencies of target tini-staticScanning dependencies of target tini

[ 50%] [100%] Building C object CMakeFiles/tini.dir/src/tini.c.oBuilding C object CMakeFiles/tini-static.dir/src/tini.c.o

<command-line>:0:0: warning: "_FORTIFY_SOURCE" redefined
<built-in>: note: this is the location of the previous definition
<command-line>:0:0: warning: "_FORTIFY_SOURCE" redefined
<built-in>: note: this is the location of the previous definition
Linking C executable tiniLinking C executable tini-static

[100%] Built target tini
```

**Comment by krallin** [comment]
This is probably because I made warnings fatal in https://github.com/krallin/tini/commit/cd23c4ce65e8c315be2bed215b3a6614e4ca7785

It looks like `-D_FORTIFY_SOURCE=2` isn't needed on alpine because it's default: http://git.alpinelinux.org/cgit/aports/plain/main/gcc/010_all_default-fortify-source.patch.

Not sure how best to address this. Thoughts? You could drop `add_definitions (-D_FORTIFY_SOURCE=2)` in `CMakeLists.txt` in a patch (which would be consistent with what Alpine does for GCC), but that's more work for you.

---

Otherwise, I could make this warning non-fatal.

**Comment by danilobuerger** [comment]
I dont know much about it, but judging the comment i would assume that building would now fail on every  distro that has fortify source enabled by default? If that is the case, i think a solution would be better suited here to not get frustrated user from distro XY posting issues about it here.

**Comment by krallin** [comment]
Fair enough. I'll add a test for it being a builtin instead.

**Comment by krallin** [comment]
@danilobuerger can you try this branch: https://github.com/krallin/tini/tree/fix-22?

(I tested it with the official Alpine Docker image and it seems to work, but I'd appreciate it if you could confirm it :) )

**Comment by danilobuerger** [comment]
Confirmed :dancer:

**Comment by krallin** [comment]
Awesome; thanks!

I'm going to try and add a check for hardening being enabled in the CI script before I merge this (to make sure we still enable fortify source outside of Alpine!)

**Comment by krallin** [comment]
I added the tests in a separate branch (blocked on https://github.com/travis-ci/apt-package-whitelist/issues/1654 though), but I merged the fix now in #23 (version 0.8.3).

Thanks for the report!
